code/modules/library/lib_machines.dm

```diff
@@ -17,23 +17,28 @@ datum/borrowbook // Datum used to keep track of who has borrowed what when and f
 	var/getdate
 	var/duedate
 
+
 /*
  * Library Public Computer
  */
-/obj/machinery/librarypubliccomp
+/obj/machinery/computer/libraryconsole
 	name = "library visitor console"
-	icon = 'icons/obj/library.dmi'
-	icon_state = "computer"
-	anchored = 1
-	density = 1
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "library"
+	circuit = /obj/item/weapon/circuitboard/libraryconsole

```

Comment:
You need to set `circuit.build_path =  /obj/machinery/computer/libraryconsole` on `New()` for the public computer, otherwise you'll get the wrong circuit board from the library public computer (meaning you can unscrew it and screw again to get a book computer).